src/shogun/machine/GeneralizedLinearMachine.h

```diff
@@ -0,0 +1,215 @@
+		 * @param src_w new w
+		 */
+		virtual void set_w(const SGVector<float64_t> src_w);
+
+		/** set bias
+		 *
+		 * @param b new bias
+		 */
+		virtual void set_bias(float64_t b);
+
+		/** get bias
+		 *
+		 * @return bias
+		 */
+		virtual float64_t get_bias() const;
+
+ 		/** set features
+ 		 *
+ 		 * @param feat features to set
+ 		 */
+ 		virtual void set_features(std::shared_ptr<DotFeatures> feat);
+
+		/**
+		* A specialization of the train_machine method
+		* @param feats training data
+		*/
+		virtual bool train_machine(std::shared_ptr<const DenseFeatures<float64_t>> feats);
+
+		/** apply linear machine to data
+		 * for regression problem
+		 *
+		 * @param data (test)data to be classified
+		 * @return classified labels
+		 */
+		virtual std::shared_ptr<RegressionLabels> apply_regression(std::shared_ptr<Features> data=NULL);
+
+ 		/** get features
+ 		 *
+ 		 * @return features
+ 		 */
+ 		virtual std::shared_ptr<DotFeatures> get_features();
+
+ 		/** Returns the name of the SGSerializable instance.  It MUST BE
+ 		 *  the CLASS NAME without the prefixed `C'.
+ 		 *
+ 		 * @return name of the SGSerializable
+ 		 */
+ 		virtual const char* get_name() const { return "GeneralizedLinearMachine"; }
+
+	protected:
+
+		/** apply get outputs
+		 *
+		 * @param data features to compute outputs
+		 * @return outputs
+		 */
+		virtual SGVector<float64_t> apply_get_outputs(std::shared_ptr<Features> data);
+
+		/**Conditional intensity function.*/
+		virtual SGVector<float64_t> conditional_intensity(SGMatrix<float64_t> X, SGVector<float64_t> w, float64_t bias);
+
+		/** predict for one vector */
+		virtual SGVector<float64_t> predict(SGMatrix<float64_t> X);
+
+		/** fit model */
+		virtual bool fit(SGMatrix<float64_t> X, SGVector<float64_t> y);
+
+		/** compute gradient of weights */
+		virtual SGVector<float64_t> compute_grad_L2_loss_w(SGMatrix<float64_t> X, SGVector<float64_t> y, SGVector<float64_t> w, float64_t bias);
+
+		/** compute gradient of bias */
+		virtual float64_t compute_grad_L2_loss_bias(SGMatrix<float64_t> X, SGVector<float64_t> y, SGVector<float64_t> w, float64_t bias);
+
+		/** compute z */
+		virtual SGVector<float64_t> compute_z(SGMatrix<float64_t> X, SGVector<float64_t> w, float64_t bias);
+
+		/** conditional non-linear function */
+		virtual SGVector<float64_t> non_linearity(SGVector<float64_t> z);
+
+		/** compute gradient of non-linearity */
+		virtual SGVector<float64_t> gradient_non_linearity(SGVector<float64_t> z);
+
+		virtual SGVector<float64_t> apply_proximal_operator(SGVector<float64_t> w, float64_t threshold);
+
+	private:
+
+		void init();
+
+	protected:
+		/** w */
+		SGVector<float64_t> m_w;
+
+		/** bias */
+		float64_t m_bias;
+
+ 		/** features */
+		std::shared_ptr<DotFeatures> features;
+
+		/** Distribution type */
+		GLM_DISTRIBUTION distribution;
+
+		/** specifies if a constant (a.k.a. bias or intercept) should be added to the decision function.
+         * default: true */
+		bool m_fit_intercept;
```

Comment:
this this is also in linear machine

Reply:
I am getting this error message?
```
class "shogun::LinearMachine" has no member "m_compute_bias"
```
Also I was unable to find any protected member variable `m_compute_bias` in `LinearMachine.h` as mentioned in the docs

Reply:
@karlnapf The way I am doing it right now is correct, right? since, as I mentioned, there is no m_compute_bias in the LinearMachine class. https://github.com/shogun-toolbox/shogun/pull/5006#discussion_r417323500

Reply:
sorry, what do you mean? I don't understand the question